sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/systemevents/AcsRouterJobClassificationFailedEventData.java

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+package com.azure.messaging.eventgrid.systemevents;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.models.ResponseError;
+import com.azure.messaging.eventgrid.implementation.models.AcsRouterCommunicationError;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Schema of the Data property of an EventGridEvent for a Microsoft.Communication.RouterJobClassificationFailed event.
+ */
+@Fluent
+public final class AcsRouterJobClassificationFailedEventData extends AcsRouterJobEventData {
+
+    /*
+     * Router Job Classification Policy Id
+     */
+    @JsonProperty(value = "classificationPolicyId")
+    private String classificationPolicyId;
+
+    /*
+     * Router Job Classification Failed Errors
+     */
+    @JsonProperty(value = "errors")
+    private List<AcsRouterCommunicationError> errors;
+
+    /** Creates an instance of AcsRouterJobClassificationFailedEventData class. */
+    public AcsRouterJobClassificationFailedEventData() {}
+
+    /**
+     * Get the classificationPolicyId property: Router Job Classification Policy Id.
+     *
+     * @return the classificationPolicyId value.
+     */
+    public String getClassificationPolicyId() {
+        return this.classificationPolicyId;
+    }
+
+    /**
+     * Set the classificationPolicyId property: Router Job Classification Policy Id.
+     *
+     * @param classificationPolicyId the classificationPolicyId value to set.
+     * @return the AcsRouterJobClassificationFailedEventData object itself.
+     */
+    public AcsRouterJobClassificationFailedEventData setClassificationPolicyId(String classificationPolicyId) {
+        this.classificationPolicyId = classificationPolicyId;
+        return this;
+    }
+
+    /**
+     * Get the errors property: Router Job Classification Failed Errors.
+     *
+     * @return the errors value.
+     */
+    public List<ResponseError> getErrors() {
+        return this.errors.stream()
+                .map(e -> new ResponseError(e.getCode(), e.getMessage()))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Set the errors property: Router Job Classification Failed Errors.
+     *
+     * @param errors the errors value to set.
+     * @return the AcsRouterJobClassificationFailedEventData object itself.
+     */
+    public AcsRouterJobClassificationFailedEventData setErrors(List<ResponseError> errors) {
+        this.errors =
+                errors.stream()
+                        .map(e -> new AcsRouterCommunicationError().setCode(e.getCode()).setMessage(e.getMessage()))
+                        .collect(Collectors.toList());
+        return this;
```

Comment:
Is this conversion needed? Can the swagger be updated to use the [common error type](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/data-plane/v1/types.json#L48) which will handle the mapping to ResponseError?

Reply:
this was discussed and there's code-gen blocks to doing this well, I gather.